### ISSUE TITLE
Client -> nginx reverse proxy -> external forward proxy -> target service possible?

### ISSUE THREAD

**Issue by BenjaminDaSilva** [chobits/ngx_http_proxy_connect_module#211]
Hi there,

I found some kind of similar questions ([166](https://github.com/chobits/ngx_http_proxy_connect_module/issues/166), [210](https://github.com/chobits/ngx_http_proxy_connect_module/issues/210)) but in details my scenario is different:

I have a client C (part of a web application) which can't be configured to use a forward proxy for API calls to a service S that can only be reached via a forward proxy F. This web app though runs on a virtual appliance which runs nginx to reverse proxy inbound connections to the web app.

Could I enable this nginx with this module to act as reverse proxy R but not pass_proxy directly to service S but via the forward proxy F?

So basically:

[ C ] ---- API Call ----> [ R ] ----- CONNECT / tunnel ----> [ F ] -----> [ S ]

**Comment by chobits** [comment]
This is technically feasible. But this module does not support this feature.

It needs a new nginx upstream module to fire CONNECT tunnel request to upstream and then proxy data flow via this tunnel to backend.

According to my nginx-C developing experience, it is troublesome to modify nginx upstream from its C core base.
